conversation_classification/kaggle/model.py

```diff
@@ -36,38 +34,86 @@
 
 # Training Params
 TRAIN_SEED = 9812 # Random seed used to initialize training
-TRAIN_STEPS = 1000 # Number of steps to take while training
 LEARNING_RATE = 0.01
 BATCH_SIZE = 120
 
-# Output Params
-TEST_OUT_PATH = 'test_out.csv' # Where to write results on heldout data
-PREDICT_OUT_PATH = 'predict_out.csv' # Where to write results on unlabled data
-
 class WikiData:
 
-  def __init__(self, path):
-    self.data = self._load_data(path)
-    self.data['comment_text'] = self.data['comment_text'].astype(str)
+  def __init__(self, data_path, y_class, vocab_processor_path=None,
+               test_mode=False, seed=None, train_percent=None):
+    """
+    Args:
+      * data_path (string): path to file containing train or test data
+      * y_class (string): the class we're training or testing on
+      * vocab_processor_path (string): if provided, the comment_text data will be
+          processed with the vocab processor at that location. If not, a new
+          vocab_processor will be created using the training data.
+      * test_mode (boolean): true if loading data just to test on, not training a model
+      * seed (integer): a random seed to use for data splitting
+      * train_percent (fload): the percent of data we should use for training data
+    """
+    data = self._load_data(data_path)
+
+    self.x_train, self.x_train_text = None, None
+    self.x_test, self.x_test_text = None, None
+    self.y_train = None
+    self.y_test = None
+    self.vocab_processor = None
+
+    # If test_mode is True, then put all the data in x_test and y_test
+    if test_mode:
+      train_percent = 0
+      seed = None
```

Comment:
Why does seed = None in test mode?

Reply:
Good point, I don't think there's a reason.